### PR TITLE
feat: create models and apis for category, shop and product

### PR DESCRIPTION
**Pull request by rutajdash**
Models Created:
- Ad Model
- Category Model
- Shop Model
- Product Model
- Review Model
- Discount Coupon Model

APIs Created:
- /user/retail/get_category/:catId
- /user/retail/get_subCategory/:catId
- /user/retail/get_shop/:shopId
- /user/retail/get_product_details/:prodId

TODOs:
- Implement trending logic

Others:
- Extracted and Refactored Address Schema

### CODE REVIEW

File: models/Address_Schema.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+module.exports = new Schema({
+	city: {
+		type: String,
+	},
+	street: {
+		type: String,
+	},
+	houseNumber: {
+		type: String,
+	},
```
Comment (dkb-startup) [inline models/Address_Schema.js:12]: Can we have an addition fields  like 
1. Pin Code
2. Landmark

File: models/Discount_Coupon_Model.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	},
+	min_order_value: {
+		type: Number,
+		required: false,
+	},
+	code: {
+		type: String,
+		required: true,
+	},
+	product: {
```
Comment (dkb-startup) [inline models/Discount_Coupon_Model.js:24]: Product may not be required for every discunt coupon as discount coupon is applied on total order value.

File: models/Address_Schema.js (hunk 1 above)
Comment (dkb-startup) [inline models/Address_Schema.js:12]: Can we have aditional field like 
1. PIN
2. Landmark
3. May be description

File: models/Product_Model.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		required: true,
+	},
+	reviews: [
+		{
+			type: schema.Types.ObjectId,
+			ref: 'Review',
+			required: false,
+		},
+	],
+	discount_coupons: [
```
Comment (dkb-startup) [inline models/Product_Model.js:49]: This might not be required.

**Review by Yash-Khatri-31 — COMMENTED** [review]
Checking 123

**Pull request closed without merging**